### PR TITLE
Simple Drag

### PR DESCRIPTION
**Pull request by sleepyquelea**
KIS has come a long way and received a lot of polish. Yet I kinda miss the simplicity of the the old "inventory" management seen in this short sequence: https://www.youtube.com/watch?v=8I7t-SrTRms&t=13m21s

So I propose two small changes:

- Allow dragging of an equipped item directly from the attached part.
- Allow dropping an item directly onto kerbals to store it in their inventory.

Quick demo: https://giant.gfycat.com/NimbleEnormousAlligatorsnappingturtle.webm

I'm not very familiar with the code base and there is probably a better way to archieve this. Please let me know if there is any interest in taking this further and any advice you have for the implementation.

There's still a few todos. In particular, is there a good way to get the corresponding KIS_Item for a ModuleKISItem? Currently I'm just comparing all items in the inventory.

I also didn't have a good way to provide user feedback. KISAddonCursor doesn't seem to be running while a part is dragged. Maybe a KISDragCursor should be added?

### CODE REVIEW

File: Source/KISAddonPickup.cs [hunk 1]
```diff
@@ -719,6 +721,24 @@ public enum PickupMode {
     KISAddonCursor.StopPartDetection();
   }
 
+  bool TryPutDraggedPartIntoInventory() {
+    // While dragging a part KISAddonCursor and KISAddonPointer are not running, so neither will work here.
+    // TODO: Would be nice to have KISAddonCursor to provide better feedback. Or maybe create something like KISDragCursor?
+    var eva = Mouse.HoveredPart?.GetComponent<KerbalEVA>();
```
Comment (ihsoft) [inline Source/KISAddonPickup.cs]: Could you, please, stick to C# 5.0 syntax? C# 6.0 has some good things (like this ?. operator), but my building system is C# 5.0. One day I'll upgrade, but it won't happen today.

**Comment by ihsoft** [comment]
The whole idea looks good and I like it! Not exactly sure about the solution, though. First of all, I don't think any equipped item should be moved like this.  In many cases people don't need equipping an item or storing it into the inventory, they want to synchronise the part and kerbal speeds. This is done exactly like this: dropping an item/part at the kerbal model. This is a very useful function when working on orbit, since with time the floating parts start drifting off the construction site. So, I'd say this "drop and equip" action should only be handled for the carriable items. Many people intuitively expect it should be working like this, so here it would be a real feature.

Another concern is the checking. I haven't tried the code yet (sorry, a bit busy with the variants support), but I feel the code should ensure that the item being unequipped can actually be equipped to the other kerbals before doing any changes to the scene. If it's already there, then good. I just haven't catched the branch when was doing a quick review.

> Please let me know if there is any interest in taking this further and any advice you have for the implementation.

I definitely think it's a good feature. However, working on it right now may result in wasted efforts. I'm doing very extensive code changes, so merging can be a pain.

>  In particular, is there a good way to get the corresponding KIS_Item for a ModuleKISItem?

Nope. The item has a reference to the module, but not the vise versa. Given the size of the kerbal backpack, enumerating doesn't seem to be a big deal. You're not doing it every single frame, right?

> I also didn't have a good way to provide user feedback. KISAddonCursor doesn't seem to be running while a part is dragged

This whole functionality is very convoluted and is not suited well for anything that is not a simple part moving. I think it's ok to present only the icon while dragging a carriable part between the krebals. Or do you mean it's not shown either?

File: Source/KISAddonPickup.cs (hunk 1 above)
Comment (sleepyquelea) [reply Source/KISAddonPickup.cs]: Sure, no problem.

**Comment by sleepyquelea** [comment]
Good idea, restricting it to carriable items makes sense. I don't often use KIS in orbit with multiple floating items. I'll do more testing here. Thanks for this insight.

> Given the size of the kerbal backpack, enumerating doesn't seem to be a big deal. You're not doing it every single frame, right?

Yes, it's only done on the initial grab. I guess it's fine then.

> Another concern is the checking.

I reused some of the code for putting an item into the inventory, so the volume check was already there and I added a check for distance to this code path. I'll do another pass here when working on better feedback.

> This whole functionality is very convoluted and is not suited well for anything that is not a simple part moving. I think it's ok to present only the icon while dragging a carriable part between the krebals. Or do you mean it's not shown either?

When dragging a part, the icon is displayed the same as it currently is. It's just that there is no functionality set up to provide feedback on the icon during this step.

If I understand correctly, KISAddonCursor only handles the icon for the initial part selection while the grab key is held down. When an item is dragged and the mouse is held down, the item icon is displayed by KISAddonPickup.OnGUI. 

This is where I would add more feedback if the receiving kerbal is too far away, doesn't have inventory space or what will actually happen if you release the mouse: part placement mode as currently or placing the carriable item into the inventory.

> However, working on it right now may result in wasted efforts. I'm doing very extensive code changes, so merging can be a pain.

Good to know. I'll play some more with this and revisit later.

**Comment by ihsoft** [comment]
I apologize for skipping this PR. Somehow, it missed my attention during the (almost) year! Will try to integrate it now.

**Comment by ihsoft** [comment]
Tried to merge this PR, but the trunk has progressed too far. So, I copied the files and manually adjusted the changes. The credits to the change are reflected here: https://github.com/ihsoft/KIS/blob/next/CHANGELOG.md.

I close this PR since it cannot be integrated, but the work was not a waste! THANKS!!!

**Pull request closed without merging**